File: cpp_make_doctest/test/test_libs.cpp
////
//// Test library
////
#include <doctest/doctest.h>

#include "libs.hpp"

TEST_CASE("libs") {
  CHECK(libs::add(1, 2) == 3);
  CHECK(libs::add(-3, 2) == -1);
  CHECK(libs::add(0, 0) == 0);
  // CHECK(libs::add(0, 0) == 1);  // invalid test case
}

File: f90_make_mintest/tests/test_libs.f90
!==============================================================================!
!          file: test_libs.f90
!   description: Test library.
!        author: H. Iijima
!          date: 2024-09-17
!==============================================================================!

#include "mintest.f90"
MINTEST_DEFINE_MAIN


subroutine test_case()
  use libs
  implicit none

  check(abs(add(1.d0, -1.d0)) < 1.d-16)
  check(abs(add(2.d0, -1.d0)) < 1.d-16)
end subroutine test_case
